### PR TITLE
[seacas/vtkm/vtk/paraview] Add/update ports. 

### LINKED ISSUE #23926: [paraview] update to v5.10.1
Library name: paraview

New version number: 5.10.1

Other information that may be useful (release notes, etc...)  
Announcement: https://discourse.paraview.org/t/paraview-5-10-1-available-for-download/9188  
Release notes: https://gitlab.kitware.com/paraview/paraview/-/blob/master/Documentation/release/ParaView-5.10.1.md

### PR DESCRIPTION
**Pull request by Neumann-A**
- add ~~trilinos~~ seacas port (since only that is used by vtk)
- add verdict port
- update vtkm
- update vtk
- update paraview to 5.11
- fix issues observed along the way 

see if linux/osx CI works. Only tested windows with `paraview[core,mpi,python,tools,vtkm]`

~~TODO: add https://gitlab.kitware.com/vtk/vtk/-/commit/bfa3e4c7621ddf5826755536eb07284c86db6474.diff~~
Fixes https://github.com/microsoft/vcpkg/issues/23926.

### CODE REVIEW

**Comment by Neumann-A** [comment]
@bartlettroscoe I mean I patched the hell out of the tribits part of trilinos here but I think there are a lot of issues. 

a) x64-linux failure in vtk is due to trilinos seemingly using the wrong (lower)case package name internally while externally they expect uppercase (maybe force it )
b) x64-osx failure is due to some externally dependency in trilinios not being found. I assume it is also due to a case issue but haven't investigated this. `Trilinos_ENABLE_ALL_PACKAGES:BOOL=OFF` partly ignored ?!?!

Other stuff:
no dynamic builds on windows: kokkos fails to build. I assume somebody played to much with some CMAKE_ flags since it complains that it cannot access *.lib (import file) file which it should generate itself, (e.g. building kokkoscore.dll should generate kokkoscore.lib but that fails with an access denied error)
no builds with kokkos enabled due to link errors while linking seacas. 
I patched out the vendored fmt stuff since it also generated link errors.

TL;DR: Usage should actually be tested by trilinos upstream

-------------------------------

@mathstuf
About the x64-windows* failures:
```
CMake Error at D:/installed/x64-windows/share/vtk/VTK-targets.cmake:954 (set_target_properties):
  The link interface of target "VTK::jsoncpp" contains:

    JsonCpp::JsonCpp

  but the target was not found.  Possible reasons include:
```
From what I see the package name seems to be wrongly cased (https://gitlab.kitware.com/vtk/vtk/-/blob/master/ThirdParty/jsoncpp/CMakeLists.txt#L11 while usage seems to indicate `find_package(jsoncpp CONFIG REQUIRED)` with `jsoncppConfig.cmake`). Bug in VTK? But why didn't it fail while building VTK? Did it use its internal version instead?

----------------------------------

Hmm for VTK it is probably better to have a seacas/zoltan(/kokkos?) port instead of a trilinos port even though the zoltan build requires to pull in all of trilinos which I commit to this route here.

**Comment by bartlettroscoe** [comment]
@Neumann-Al

> a) x64-linux failure in vtk is due to trilinos seemingly using the wrong (lower)case package name internally while externally they expect uppercase (maybe force it )

Can you create a Trilinos GitHub issue for this?

> b) x64-osx failure is due to some externally dependency in trilinios not being found. I assume it is also due to a case issue but haven't investigated this. `Trilinos_ENABLE_ALL_PACKAGES:BOOL=OFF` partly ignored ?!?!

`Trilinos_ENABLE_ALL_PACKAGES` has no direct impact on external dependencies.

Can you create a Trilinos GitHub issue for this?

If things are only getting resolved in external patches of Trilinos that will not impact future versions of Trilinos.

> no dynamic builds on windows: kokkos fails to build.

Have you contacted the Kokkos developers about their support for Windows?  I think you can do that with posting an issue in github.com/kokkos/kokkos.

Trilinos needs to decide on the level of support for Windows.  (It is unclear to me what if any support for Windows exists.)

File: ports/trilinos/fix_tpl_libs.patch [hunk 1]
```diff
@@ -0,0 +1,264 @@
+diff --git a/cmake/tribits/common_tpls/FindTPLNetcdf.cmake b/cmake/tribits/common_tpls/FindTPLNetcdf.cmake
+index f35dc8914..27e395545 100644
+--- a/cmake/tribits/common_tpls/FindTPLNetcdf.cmake
++++ b/cmake/tribits/common_tpls/FindTPLNetcdf.cmake
+@@ -70,19 +70,19 @@ if (Netcdf_ALLOW_PREFIND)
+     "${CMAKE_CURRENT_LIST_DIR}/utils"
+      )
+ 
+-  find_package(NetCDF)
++  find_package(netCDF CONFIG)
```
Comment (bartlettroscoe) [inline ports/trilinos/fix_tpl_libs.patch:10]: I think this conflicts with changes made on Trilinos 'develop' and this issue is already resolved there.  See:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/tribits/common_tpls/FindTPLNetcdf.cmake

Comment (Neumann-A) [reply ports/trilinos/fix_tpl_libs.patch:10]: I think i can remove this change. It was before i noticed the line `if (Netcdf_ALLOW_MODERN AND HDF5_FOUND_MODERN_CONFIG_FILE)` blocking the `CONFIG` lookup.

This PR is still WIP and I haven't cleaned up the patches yet. First thing to do is make CI happy and after that do the cleanup.

**Comment by mathstuf** [comment]
> From what I see the package name seems to be wrongly cased (https://gitlab.kitware.com/vtk/vtk/-/blob/master/ThirdParty/jsoncpp/CMakeLists.txt#L11 while usage seems to indicate find_package(jsoncpp CONFIG REQUIRED) with jsoncppConfig.cmake). Bug in VTK? But why didn't it fail while building VTK? Did it use its internal version instead?

Yes. It seems that upstream diverged from the spelling of the imported targets made in `FindJsonCpp` that VTK has. We'll have to add compat bits and migrate.

**Comment by Neumann-A** [comment]
OSX error (although ParMETIS is explicitly disabled): 
```
/Users/vagrant/Data/buildtrees/trilinos/src/c0e86bcc81-dff7f2d0eb.clean/packages/zoltan/src/tpls/third_library.c:111:10: error: implicit declaration of function 'Zoltan_ParMetis' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    rc = Zoltan_ParMetis(zz, part_sizes, num_imp, imp_gids, imp_lids,
         ^
1 error generated.
```
Hmm maybe the logic `IF (${PACKAGE_NAME}_ENABLE_ParMETIS OR ${PACKAGE_NAME}_ENABLE_METIS)` is a bit misleading or i need to disable werror. hmm no `third_party.c` is just missing an include....

File: ports/vtk-dicom/portfile.cmake [hunk 2]
```diff
@@ -1,7 +1,7 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO dgobbi/vtk-dicom
-    REF 5034c68450de857b70fbe4a4b9f8dddb62badef3 # v0.8.12
+    REF 5eff7f1df7b7cd31ef95b5059c2774d3d71a2d0e # v0.8.13
     SHA512 bad1ed6a4a412402a2cd69e5f85b2b73f1ee7ea46a6bbcac31c5f66d07ae006679ffbd9a3c70f9baa1b05b1af0a2d4ca0efc34ec0a85a92f5116b900e81635cd
```
Comment (Neumann-A) [inline ports/vtk-dicom/portfile.cmake:5]: cc @BillyONeal: Please explain why this didn't threw a hash error. I was expecting a hash error here. 
```
Building vtk-dicom[core]:x64-linux...
-- Downloading https://github.com/dgobbi/vtk-dicom/archive/5eff7f1df7b7cd31ef95b5059c2774d3d71a2d0e.tar.gz -> dgobbi-vtk-dicom-5eff7f1df7b7cd31ef95b5059c2774d3d71a2d0e.tar.gz...
-- Extracting source /mnt/vcpkg-ci/downloads/dgobbi-vtk-dicom-5eff7f1df7b7cd31ef95b5059c2774d3d71a2d0e.tar.gz
-- Applying patch std.patch
-- Using source at /mnt/vcpkg-ci/buildtrees/vtk-dicom/src/d3d71a2d0e-2fd6b59f41.clean
-- Configuring x64-linux-dbg
```

**Comment by Neumann-A** [comment]
@mathstuf: after reviewing the trace logs for `rtabmap` it looks like VTK is not doing a `find_(package|dependency
)(JsonCpp|jsoncpp)` call.  Any idea what might be the cause for that behavior?

**Comment by mathstuf** [comment]
The only reasons I can think of are:

- the jsoncpp module is disabled;
- using the vendored copy.

Otherwise it should be searching at build and `find_package(VTK)` time.

**Comment by Neumann-A** [comment]
@dg0yt: Hidden `libssh` dep in the `opencv` build via `FFMPEG`? (just pinging you because you touched opencv4 deps last)

```
CMake Error at /mnt/vcpkg-ci/installed/x64-linux/share/ffmpeg/FindFFMPEG.cmake:68 (find_library):
  Could not find FFMPEG_DEPENDENCY_ssh_RELEASE using the following names: ssh
Call Stack (most recent call first):
```

**Comment by Neumann-A** [comment]
++times_i_cannot_reproduce_a_problem_in_ci_locally:  `rtabmap` missing `JsonCpp::JsonCpp` and not even looking for it via `find_package`

**Comment by Neumann-A** [comment]
rtabmap regression is probably a cmake bug fixed in cmake 3.24.1 which is why I couldn't reproduce it locally.

**Comment by codehasbug** [comment]
Hi! @Neumann-A , I found the VTK branch "66143ef041b980a51e41ee470d053e67209150f8" has problem. When I build it by MSVC2019, it failed. This is due to `vtkStdString::operator const char*` . Please check this link to found detail. [VTK failed to build with error 2666 on MSVC](https://gitlab.kitware.com/vtk/vtk/-/issues/18672). I think you could change it to a new branch. Have a good day.

**Comment by mathstuf** [comment]
Backporting https://gitlab.kitware.com/vtk/vtk/-/merge_requests/9572 might help there.

**Comment by Neumann-A** [comment]
Already on the TODO. Waiting for #28012 to be merged.

**Comment by Neumann-A** [comment]
rtabmap target dependency call chain for `JsonCpp::JsonCpp`

```
D:/buildtrees/rtabmap/src/fde13e9126-29a106af5b.clean/corelib/src/CMakeLists.txt(737):  TARGET_LINK_LIBRARIES(rtabmap_core rtabmap_utilite opencv_calib3d;opencv_core;opencv_dnn;opencv_features2d;opencv_flann;opencv_gapi;opencv_highgui;opencv_imgcodecs;opencv_imgproc;opencv_ml;opencv_objdetect;opencv_photo;opencv_stitching;opencv_video;opencv_videoio;opencv_alphamat;opencv_aruco;opencv_barcode;opencv_bgsegm;opencv_bioinspired;opencv_ccalib;opencv_cudaarithm;opencv_cudabgsegm;opencv_cudacodec;opencv_cudafeatures2d;opencv_cudafilters;opencv_cudaimgproc;opencv_cudalegacy;opencv_cudaobjdetect;opencv_cudaoptflow;opencv_cudastereo;opencv_cudawarping;opencv_cudev;opencv_cvv;opencv_datasets;opencv_dnn_objdetect;opencv_dnn_superres;opencv_dpm;opencv_face;opencv_freetype;opencv_fuzzy;opencv_hdf;opencv_hfs;opencv_img_hash;opencv_intensity_transform;opencv_line_descriptor;opencv_mcc;opencv_optflow;opencv_ovis;opencv_phase_unwrapping;opencv_plot;opencv_quality;opencv_rapid;opencv_reg;opencv_saliency;opencv_sfm;opencv_shape;opencv_stereo;opencv_structured_light;opencv_superres;opencv_surface_matching;opencv_text;opencv_tracking;opencv_videostab;opencv_viz;opencv_wechat_qrcode;opencv_xfeatures2d;opencv_ximgproc;opencv_xobjdetect;opencv_xphoto;pcl_common;pcl_octree;pcl_io;pcl_kdtree;pcl_search;pcl_surface;pcl_filters;pcl_features;pcl_registration;pcl_sample_consensus;pcl_ml;pcl_segmentation;optimized;D:/installed/x64-windows/lib/boost_system-vc140-mt.lib;debug;D:/installed/x64-windows/debug/lib/boost_system-vc140-mt-gd.lib;optimized;D:/installed/x64-windows/lib/boost_filesystem-vc140-mt.lib;debug;D:/installed/x64-windows/debug/lib/boost_filesystem-vc140-mt-gd.lib;optimized;D:/installed/x64-windows/lib/boost_date_time-vc140-mt.lib;debug;D:/installed/x64-windows/debug/lib/boost_date_time-vc140-mt-gd.lib;optimized;D:/installed/x64-windows/lib/boost_iostreams-vc140-mt.lib;debug;D:/installed/x64-windows/debug/lib/boost_iostreams-vc140-mt-gd.lib;optimized;D:/installed/x64-windows/lib/boost_serialization-vc140-mt.lib;debug;D:/installed/x64-windows/debug/lib/boost_serialization-vc140-mt-gd.lib;optimized;D:/installed/x64-windows/lib/boost_regex-vc140-mt.lib;debug;D:/installed/x64-windows/debug/lib/boost_regex-vc140-mt-gd.lib;VTK::ChartsCore;VTK::CommonColor;VTK::CommonComputationalGeometry;VTK::CommonCore;VTK::CommonDataModel;VTK::CommonExecutionModel;VTK::CommonMath;VTK::CommonMisc;VTK::CommonTransforms;VTK::FiltersCore;VTK::FiltersExtraction;VTK::FiltersGeneral;VTK::FiltersGeometry;VTK::FiltersModeling;VTK::FiltersSources;VTK::ImagingCore;VTK::ImagingSources;VTK::InteractionStyle;VTK::InteractionWidgets;VTK::IOCore;VTK::IOGeometry;VTK::IOImage;VTK::IOLegacy;VTK::IOPLY;VTK::RenderingAnnotation;VTK::RenderingCore;VTK::RenderingContext2D;VTK::RenderingLOD;VTK::RenderingFreeType;VTK::ViewsCore;VTK::ViewsContext2D;VTK::RenderingOpenGL2;VTK::RenderingQt;VTK::GUISupportQt;FLANN::FLANN;optimized;D:/installed/x64-windows/lib/zlib.lib;debug;D:/installed/x64-windows/debug/lib/zlibd.lib;D:/installed/x64-windows/debug/lib/sqlite3.lib;OpenMP::OpenMP_CXX )
D:/installed/x64-windows/share/opencv/OpenCVModules-debug.cmake(141):  set_target_properties(opencv_viz PROPERTIES IMPORTED_IMPLIB_DEBUG D:/installed/x64-windows/debug/lib/opencv_viz4d.lib IMPORTED_LINK_DEPENDENT_LIBRARIES_DEBUG VTK::FiltersExtraction;VTK::FiltersSources;VTK::FiltersTexture;VTK::IOExport;VTK::IOGeometry;VTK::IOPLY;VTK::InteractionStyle;VTK::RenderingCore;VTK::RenderingLOD;VTK::RenderingOpenGL2 IMPORTED_LOCATION_DEBUG D:/installed/x64-windows/debug/bin/opencv_viz4d.dll )
D:/installed/x64-windows/share/vtk/VTK-targets.cmake(557):  set_target_properties(VTK::IOExport PROPERTIES INTERFACE_INCLUDE_DIRECTORIES D:/installed/x64-windows/include/vtk-9.2 INTERFACE_LINK_LIBRARIES VTK::CommonCore;VTK::IOCore;VTK::IOImage;VTK::IOXML;VTK::RenderingContext2D;VTK::RenderingCore;VTK::RenderingFreeType;VTK::RenderingVtkJS )
D:/installed/x64-windows/share/vtk/VTK-targets.cmake(519):  set_target_properties(VTK::RenderingVtkJS PROPERTIES INTERFACE_INCLUDE_DIRECTORIES D:/installed/x64-windows/include/vtk-9.2 INTERFACE_LINK_LIBRARIES VTK::CommonCore;VTK::RenderingSceneGraph;VTK::jsoncpp )
D:/installed/x64-windows/share/vtk/VTK-targets.cmake(444):  set_target_properties(VTK::jsoncpp PROPERTIES INTERFACE_LINK_LIBRARIES JsonCpp::JsonCpp )
```

error is still:
```
CMake Error at D:/installed/x64-windows/share/vtk/VTK-targets.cmake:444 (set_target_properties):
  The link interface of target "VTK::jsoncpp" contains:

    JsonCpp::JsonCpp

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  D:/installed/x64-windows/share/vtk/vtk-config.cmake:136 (include)
  C:/a/2/s/scripts/buildsystems/vcpkg.cmake:829 (_find_package)
  D:/installed/x64-windows/share/pcl/PCLConfig.cmake:274 (find_package)
  D:/installed/x64-windows/share/pcl/PCLConfig.cmake:323 (find_VTK)
  D:/installed/x64-windows/share/pcl/PCLConfig.cmake:523 (find_external_library)
  C:/a/2/s/scripts/buildsystems/vcpkg.cmake:829 (_find_package)
  CMakeLists.txt:227 (FIND_PACKAGE)

```

trace logs at: https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/81668/artifacts?artifactName=failure%20logs%20for%20x64-windows&api-version=7.0&%24format=zip

**Comment by cenit** [comment]
did you already fix the error you pinged me for in https://github.com/microsoft/vcpkg/pull/27928/commits/68f970dece9c7264f991421e4e2affc114c851ef?
if so, great :)
if not, please tell me

**Comment by Neumann-A** [comment]
> did you already fix the error you pinged me for in https://github.com/microsoft/vcpkg/commit/68f970dece9c7264f991421e4e2affc114c851ef?

Already fixed. The whole `find_dependency` stuff was behind a `if(VCPKG_LIBRARY_LINKAGE STRQEUAL "static)` in `opencv4` for some reason. That `rtabmap` build in the current baseline is because `find_package(VTK COMPONENTS <comp>)` currently pulls in everything and not just `<comp>`. The VTK update changed that so that now the failure of `opencv4` to actually call `find_dependency(VTK)` gets exposed. It was a bit of difficult to debug since `find_package(VTK COMPONENTS <comp>)` still pulls in all the `VTK::` targets but leaves targets not requested in a unusable state by not pulling in their required dependencies. I was informed that people complaint of `find_package(VTK)` being slow so they optimized it if not all components are required.

File: ports/openimageio/vcpkg.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "openimageio",
   "version": "2.3.17.0",
-  "port-version": 4,
+  "port-version": 6,
```
Comment (Cheney-W) [inline ports/openimageio/vcpkg.json:4]:
```suggestion
  "port-version": 5,
```

File: versions/o-/openimageio.json [hunk 4]
```diff
@@ -1,5 +1,15 @@
 {
   "versions": [
+    {
+      "git-tree": "72b55671226434cca53fe0dea5de79b80c39d714",
+      "version": "2.3.17.0",
+      "port-version": 6
+    },
```
Comment (Cheney-W) [inline versions/o-/openimageio.json:7]:
```suggestion
```

File: ports/hdf5/portfile.cmake [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# highfive should be updated together with hdf5
+ # highfive should be updated together with hdf5
```
Comment (Cheney-W) [inline ports/hdf5/portfile.cmake:1]:
```suggestion
# highfive should be updated together with hdf5
```

File: scripts/test_ports/vcpkg-ci-paraview/vcpkg.json [hunk 6]
```diff
@@ -4,6 +4,7 @@
   "port-version": 3,
```
Comment (Cheney-W) [inline scripts/test_ports/vcpkg-ci-paraview/vcpkg.json:4]:
```suggestion
  "port-version": 4,
```

**Comment by Neumann-A** [comment]
@Cheney-W: Still reviewing/testing?

**Comment by Neumann-A** [comment]
@vicroms status of this PR?

**Comment by vicroms** [comment]
I'm halfway through reviewing it. I'll finish reviewing later today.

**Review by vicroms — DISMISSED** [review]

File: ports/vtk-m/portfile.cmake [hunk 7]
```diff
@@ -40,17 +42,19 @@ list(APPEND OPTIONS -DBUILD_TESTING=OFF)
...
   SOURCE_PATH "${SOURCE_PATH}"
   OPTIONS ${OPTIONS}
 )
 vcpkg_cmake_install()
-vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/vtkm-1.6 PACKAGE_NAME vtkm)
+vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/vtkm-${VERSION_MAJOR_MINOR}" PACKAGE_NAME vtkm)
 
-vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/vtkm/VTKmConfig.cmake" [[set_and_check(VTKm_CONFIG_DIR "${PACKAGE_PREFIX_DIR}/lib/cmake/vtkm-1.6")]] [[set_and_check(VTKm_CONFIG_DIR "${PACKAGE_PREFIX_DIR}/share/vtkm")]])
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/vtkm/VTKmConfig.cmake" 
+                    [[set_and_check(VTKm_CONFIG_DIR "${PACKAGE_PREFIX_DIR}/lib/cmake/vtkm-1.9")]]
```
Comment (Osyotr) [inline ports/vtk-m/portfile.cmake:56]:
```suggestion
                    [[set_and_check(VTKm_CONFIG_DIR "${PACKAGE_PREFIX_DIR}/lib/cmake/vtkm-${VERSION_MAJOR_MINOR}")]]
```

File: ports/vtk-m/portfile.cmake [hunk 8]
```diff
@@ -9,6 +9,8 @@
 # add_subdirectory(thirdparty/taotuple)
 # add_subdirectory(thirdparty/lcl)
 
+set(VERSION_MAJOR_MINOR 1.9) # Make sure to also change line 56
```
Comment (Osyotr) [inline ports/vtk-m/portfile.cmake:12]: Why not use it directly? And maybe also parse `VERSION` for this?

File: ports/vtk/portfile.cmake [hunk 9]
```diff
@@ -17,21 +25,18 @@ vcpkg_from_github(
         pegtl.patch
         pythonwrapper.patch # Required by ParaView to Wrap required classes
         NoUndefDebug.patch # Required to link against correct Python library depending on build type.
-        python_debug.patch
```
Comment (Osyotr) [inline ports/vtk/portfile.cmake:20]: This patch is not used anymove, shoud be deleted

File: ports/paraview/portfile.cmake [hunk 10]
```diff
@@ -38,54 +33,58 @@ vcpkg_from_gitlab(
...
 file(COPY "${VISITIT_SOURCE_PATH}/" DESTINATION "${SOURCE_PATH}/Utilities/VisItBridge")
 file(COPY "${QTTESTING_SOURCE_PATH}/" DESTINATION "${SOURCE_PATH}/ThirdParty/QtTesting/vtkqttesting")
-file(COPY "${CATALYST_SOURCE_PATH}/" DESTINATION "${SOURCE_PATH}/ThirdParty/catalyst/vtkcatalyst/catalyst")
+file(COPY "${ICET_SOURCE_PATH}/" DESTINATION "${SOURCE_PATH}/ThirdParty/IceT/vtkicet")
 
 if("python" IN_LIST FEATURES)
-    vcpkg_find_acquire_program(PYTHON3)
+    set(python_ver "")
+    if(NOT VCPKG_TARGET_IS_WINDOWS)
+        set(python_ver 3.10)
```
Comment (Osyotr) [inline ports/paraview/portfile.cmake:67]: Can this not be hardcoded?

Comment (Neumann-A) [reply ports/paraview/portfile.cmake:67]: i would like not hardcoding it but don't know how. everything I come up with requires the version somehow. and other ports also hardcode this.

**Comment by Neumann-A** [comment]
Hmm whats up with those regressions, they are all ZSTD related.

**Comment by dg0yt** [comment]
> Hmm whats up with those regressions, they are all ZSTD related.

Did gdal 3.6.2 bring a new find module?

~~~
CMake Error at /mnt/vcpkg-ci/downloads/tools/cmake-3.25.0-linux/cmake-3.25.0-linux-x86_64/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find ZSTD (missing: ZSTD_LIBRARY) (found version "1.5.2")
Call Stack (most recent call first):
  /mnt/vcpkg-ci/downloads/tools/cmake-3.25.0-linux/cmake-3.25.0-linux-x86_64/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /mnt/vcpkg-ci/installed/x64-linux/share/gdal/packages/FindZSTD.cmake:68 (find_package_handle_standard_args)
  /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:843 (_find_package)
~~~

**Comment by Neumann-A** [comment]
> Did gdal 3.6.2 bring a new find module?

Seems like it.

**Comment by Neumann-A** [comment]
https://github.com/OSGeo/gdal/blob/master/cmake/modules/packages/FindZSTD.cmake

**Comment by dg0yt** [comment]
> Did gdal 3.6.2 bring a new find module?

No, but the interaction is strange. It even finds ZSTD two lines earlier. (Log from theia.)

File: ports/paraview/portfile.cmake (hunk 10 above)
Comment (Osyotr) [reply ports/paraview/portfile.cmake:67]: Maybe parse `python3.xy-config` file and get the version from it?

**Comment by Neumann-A** [comment]
The tiff wrapper is calling the wrong FindZSTD because it APPENDs to CMAKE_MODULE_PATH instead of inserting in the front.

**Review by vicroms — APPROVED** [review]
LGTM

**Comment by Neumann-A** [comment]
@vicroms reviewed tag ?

**Review by Cheney-W — APPROVED** [review]

File: ports/seacas/portfile.cmake [hunk 11]
```diff
@@ -0,0 +1,118 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO  sandialabs/seacas
+    REF 47120843900fd7ef845688fa145ebf76a825bc51
+    SHA512 13677746457edbd4b3619576a6c474f5d8ab2eb24f648fac687e655e3121282b62994575723d18db8d18b42266d219aa4d83344ecff53f5e9a737513a3461180
+    HEAD_REF master
+    PATCHES fix_tpl_libs.patch
+            fix-ioss-includes.patch
+            deps-and-shared.patch
+            fix-mpi.patch
```
Comment (BillyONeal) [inline ports/seacas/portfile.cmake:10]: I am really Really not a fan of the amount of patching going on here. Is there a reason this needs to support being built as a DLL if upstream does not care? I would really like to see it submitted upstream at a minimum.

File: ports/seacas/portfile.cmake [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+set(enabled_projects Ioss Nemesis Exodus)
+set(proj_options "")
+foreach(proj IN LISTS disabled_projects)
+    list(APPEND proj_options "-DSeacas_ENABLE_SEACAS${proj}:BOOL=OFF")
+endforeach()
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        #--trace-expand
```
Comment (BillyONeal) [inline ports/seacas/portfile.cmake:60]:
```suggestion
```

File: ports/seacas/portfile.cmake [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+        ${FEATURE_OPTIONS}
+        -DBUILD_TESTING=OFF
+        -DSeacas_ENABLE_Zoltan:BOOL=OFF
+        -DSeacas_ENABLE_SEACAS:BOOL=ON
+        "-DSeacas_HOSTNAME:STRING=localhost"
+        "-DSeacas_GENERATE_REPO_VERSION_FILE:BOOL=OFF"
+        "-DNetcdf_ALLOW_MODERN:BOOL=ON"
+        "-DSeacas_ENABLE_Fortran:BOOL=OFF"
+        #"-DCGNS_ALLOW_PREDEFIND:BOOL=NO"
+        #"-DSeacas_ENABLE_ALL_PACKAGES:BOOL=ON"
```
Comment (BillyONeal) [inline ports/seacas/portfile.cmake:70]:
```suggestion
```

**Review by BillyONeal — COMMENTED** [review]
I am super not a fan of merging `seacas` in the condition where there are more than a thousand lines of not submitted to upstream patches, and it would be better to add these new paraview features as their own PR rather than as an update like this. However, I understand you've been jerked around here a lot by merge conflicts and stuff and the changes seem ... at least reasonable, so I'm going to merge this.

Please do look at submitting the seacas changes upstream.

seacas:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [ ] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)  
  
I wasn't really able to verify this; unfortunately "find_package" is named over 200 times in this port. Most of the examples I found did say REQUIRED so I'm willing to take your word for it for now
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

verdict
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

File: ports/seacas/portfile.cmake (hunk 11 above)
Comment (Neumann-A) [reply ports/seacas/portfile.cmake:10]: as far as I can tell from the gitblame the DLL stuff in seacas was upstreamed by mathstuf for VTK but only for the without MPI case. This is probably due to the case that MPI in conjunction with os windows is found rarely.

Comment (mathstuf) [reply ports/seacas/portfile.cmake:10]: The VTK vendoring probably doesn't enable the MPI support, so it was hidden from my patching. FWIW, `-fvisibility=hidden` would show the same problem(s) on Linux and macOS if anyone wants to track it down (I suspect VTK might want MPI-aware IOSS at some point, so "me" is certainly a possible answer there).

**Comment by petersteneteg** [comment]
Great work!!! thanks so much for doing this!
